### ISSUE TITLE
Clicking eye icon to toggle visibility does not call _set()

### ISSUE THREAD

**Issue by djrain** [godotengine/godot#42782]
**Godot version:**
3.2.2 stable

**OS/device including version:**
macOS Mojave, Macbook Pro 2015

**Issue description:**
I wish to run some code when the visibility is toggled by clicking the eye icon in the scene panel. So I tried the following code, which is the only way I'm aware of to emulate setters for builtin variables:

```
extends Node2D
tool

func _set(property, value):
	print("set something")
	match property:
		"visible":
			print("you set visibility")
```

The string is printed if the "visible" checkbox is toggled in the inspector panel, but not when clicking the eye icon as expected.

**Steps to reproduce:**
Put the above script on a Node2D and try toggling its visibility by clicking the eye.

**Minimal reproduction project:**
Don't think this is needed.

**Comment by Calinou** [comment]
Out of curiosity, is the setter called when you call `hide() or `show()` on the node *externally*?

**Comment by djrain** [comment]
> Out of curiosity, is the setter called when you call `hide()` or `show()` on the node _externally_? Maybe it's due to the editor using `hide()` or `show()` instead of setting the `visible` property.

Nope, the setter is not called. And it appears the editor is not calling those either, if I put a print inside hide() it doesn't happen unless I call it myself.

**Comment by KoBeWi** [comment]
Why not just use `NOTIFICATION_VISIBILITY_CHANGED`?

**Comment by djrain** [comment]
NOTIFICATION_VISIBILITY_CHANGED works in my case, yeah. But I still think this inconsistency should be addressed.
